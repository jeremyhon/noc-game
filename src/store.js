import { createStore } from 'redux'
import rootReducer from './reducer'

const startStore = (initialState) => {
  return createStore(
    rootReducer,
    initialState,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
}

const store = startStore()

export default store
