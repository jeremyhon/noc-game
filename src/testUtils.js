import { mount, shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import PropTypes from 'prop-types';

// enzyme MUI Test Helpers
// - https://github.com/callemall/material-ui/issues/4664

const muiTheme = getMuiTheme();

/**
* MuiMountWithContext
*
* For `mount()` full DOM rendering in enzyme.
* Provides needed context for mui to be rendered properly
* during testing.
*
* @param {obj}    node - ReactElement with mui as root or child
* @return {obj}   ReactWrapper (http://airbnb.io/enzyme/docs/api/ReactWrapper/mount.html)
*/
export const muiMount = node => mount(node, {
  context: { muiTheme },
  childContextTypes: { muiTheme: PropTypes.object },
});

/**
* MuiShallowWithContext
*
* For `shallow()` shallow rendering in enzyme (component only as a unit).
* Provides needed context for mui to be rendered properly
* during testing.
*
* @param {obj}     node - ReactElement with mui
* @return {obj}    ShallowWrapper (http://airbnb.io/enzyme/docs/api/ShallowWrapper/shallow.html)
*/
export const muiShallow = node => shallow(node, {
  context: { muiTheme },
  childContextTypes: { muiTheme: PropTypes.object },
});

// touch tap event Helpers

export const simulateTouchTap = (element) => {
  const node  = ReactDOM.findDOMNode(element.node)
  TestUtils.Simulate.touchTap(node)
}
