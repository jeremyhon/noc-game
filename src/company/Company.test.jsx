import React from 'react'
import { shallow, mount } from 'enzyme'

import { muiMount } from '../testUtils'
import store from '../store'
import ConnectedCompany, { Company } from './Company'

let component

describe('Unconnected Company', () => {
  let props

  beforeEach(() => {
    props = {
      companies: [],
      populateCompanies: jest.fn()
    }
  })

  it("matches the snapshot", () => {
    component = shallow(<Company {...props} />)
    expect(component).toMatchSnapshot()
  })

  it("generates new companies on startup", () => {
    component = mount(<Company {...props} />)
    expect(props.populateCompanies).toHaveBeenCalled()
  })
})

describe('ConnectedCompany', () => {
  beforeEach(() => {

  })

  it("matches the snapshot", () => {
    component = shallow(<ConnectedCompany store={store}/>)
    expect(component).toMatchSnapshot()
  })

  it("generates new companies on startup", () => {
    jest.spyOn(store, 'dispatch')
    component = muiMount(<ConnectedCompany store={store} />)
    expect(store.dispatch).toHaveBeenCalledWith({
      type: 'POPULATE_COMPANIES',
      payload: expect.anything(),
    })
    expect(component.find('CompanyCard')).toBePresent()
  })
})
