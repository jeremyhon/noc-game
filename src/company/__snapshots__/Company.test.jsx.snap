// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="company"
>
    
</div>,
  "nodes": Array [
    <div
      className="company"
>
      
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Company
        companies={Array []}
        addCompany={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Company {
        "_reactInternalInstance": [Circular],
        "componentDidMount": [Function],
        "context": Object {},
        "props": Object {
          "companies": Array [],
          "addCompany": [Function],
        },
        "refs": Object {},
        "renderCompanies": [Function],
        "renderCompany": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="company"
>
          
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="company"
>
          
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Company
    companies={Array []}
    addCompany={[Function]}
/>,
}
`;
